### DISCUSSION TITLE
sanboot selects removable bootable media in EFI Systems/Feature request to skip/filter removable media in sanboot

### DISCUSSION THREAD

**Discussion by darksidemilk** [ipxe/ipxe#1255] [General]
When using something like 

```
sanboot --no-describe --drive 0 
```

If a bootable CD or USB exists, it is selected as 0x80 and gets boot.

I realize that some people may find this desirable, but it can cause issues for some if that happens to be media that runs an automated install of an OS and wipes the computer.

Could there perhaps be a way to specify skipping removable media with a sanboot switch?
I realize you can get more specific with `--filename` and `--extra` but  I'm looking to have a more universal command to boot to hard drive out of the pxe menu even in multi-os environments. I know that there's the example of `--extra \EFI\{distroname}` for finding linux boot files but if I need support for any given distro while skipping removable media I end up with a difficult to maintain command, unless I'm missing something.

**Comment by NiKiZe** [comment]
Try avoid creating issues unless there is a clear bug.

iPXE has a limited set of features to chain to disk. There is no way of knowing which order anything is (in efi mode), as I think was explained in the commits that you previously also linked to.

Sure there could be something like `--exclude-device-if-device-path-contains USB` but there is no guarantee that a firmware uses that path, or doesn't also expose that device using some other path.

Most installation media has some kind of confirmation dialog before wiping, and there is nothing that prevents you from having other destructive things done with iPXE, intentionally or unintentionally. Or even having this boot directly from firmware if USB is set to first device or similar.

**Comment by darksidemilk** [comment]
Apologies, I wasn't familiar with the github discussions option.
That's a fair assessment, and what I figured, but figured it would be worth asking.
